romancal/tweakreg/tests/test_tweakreg.py

```diff
@@ -1071,17 +1100,24 @@ def test_tweakreg_handles_multiple_groups(tmp_path, base_image):
 
     res = trs.TweakRegStep.call([img1, img2])
 
-    assert len(res.models_grouped) == 2
-    all(
-        (
-            r.meta.group_id.split("-")[1],
-            i.meta.observation.program.split("-")[1],
-        )
-        for r, i in zip(res, [img1, img2])
-    )
-
-
-def test_tweakreg_multiple_groups_valueerror(tmp_path, base_image):
+    assert len(res.group_names) == 2
+    with res:
+        for r, i in zip(res, [img1, img2]):
+            assert (
+                r.meta.group_id.split("-")[1]
+                == i.meta.observation.program.split("-")[1]
+            )
+            res.shelve(r, modify=False)
+
+
+# FIXME: this test previously passed only because Tweakreg overwrote
+# the model.meta.group_id using the output from _common_name which incorrectly
+# ignored the setting of "program" in this test. This meant that tweakreg
+# found 2 groups for this test data (because ModelContainer.models_grouped)
+# worked as intended but tweakreg then overwrote the grouping and then
+# incorrectly skipped itself.
+@pytest.mark.skip(reason="This test previously incorrectly passed")
```

Comment:
I opened in issue to track this skip:
https://github.com/spacetelescope/romancal/issues/1304

Reply:
I'm marking this as resolved as based on https://github.com/spacetelescope/romancal/pull/1301 this test will be deleted. The skip is necessary here as this PR fixes the bug that is allowing it to pass.